File: frontend/app/service/note.service.js
app.service("NoteService", function($http) {
    var _getNotesByCourse = function (codCourse) {
        return $http.get("http://localhost:8080/notesharing/api/notescourse/"+codCourse);
    }

    var _getNotesByUser = function (codUser) {
        return $http.get("http://localhost:8080/notesharing/api/notesuser/"+codUser);
    }

    var _getNote = function (codNote) {
        return $http.get("http://localhost:8080/notesharing/api/note/"+codNote);
    }

    var _addNote = function (note) {
        return $http.post("http://localhost:8080/notesharing/api/addnote", note);
    }

    return {
        getNotesByCourse: _getNotesByCourse,
        getNotesByUser: _getNotesByUser,
        getNote: _getNote,
        addNote: _addNote
    }
});

File: backend/src/main/java/project/noteapp/service/NoteService.java
package project.noteapp.service;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import project.noteapp.bean.Note;
import project.noteapp.bean.NoteDTO;
import project.noteapp.dao.NoteDAO;

import java.util.List;

@Service
public class NoteService {

    @Autowired
    NoteDAO noteDAO;

    public List<NoteDTO> getNotesByCourse(String cod_course) {
        return noteDAO.getNotesByCourse(cod_course);
    }

    public NoteDTO getNote(String cod_note) {
        return noteDAO.getNote(cod_note);
    }

    public Integer createNote(NoteDTO note) {
        return noteDAO.createNote(note);
    }
}